pdns/ws-auth.cc

```diff
@@ -2029,23 +2036,37 @@ static void patchZone(UeberBackend& B, HttpRequest* req, HttpResponse* resp) {
 
         if (replace_records) {
           bool ent_present = false;
+          map<DNSName, vector<DNSResourceRecord> > records;
+
           di.backend->lookup(QType(QType::ANY), qname, di.id);
-          DNSResourceRecord rr;
-          while (di.backend->get(rr)) {
-            if (rr.qtype.getCode() == QType::ENT) {
-              ent_present = true;
-              /* that's fine, we will override it */
-              continue;
+          getAllRecords(di.backend, records);
+
+          for(auto& existing_rrset: records) {
+            bool dname_seen = false, ns_seen = false;
+            sort(existing_rrset.second.begin(), existing_rrset.second.end(),
+              [](const DNSResourceRecord& rec_a, const DNSResourceRecord& rec_b) -> bool {
+              /* we need _strict_ weak ordering */
+                return std::tie(rec_a.qname, rec_a.qtype, rec_a.content) < std::tie(rec_b.qname, rec_b.qtype, rec_b.content);
+              }
+            );
+            for(const auto& rr: existing_rrset.second) {
```

Comment:
Without the sorting a normal get() will do in this loop and will avoid some copying of rrs.

Reply:
We agreed to add the getAllRecords here so it might eventually maybe replace get().